### ISSUE TITLE
Config parameters not being saved from actual parameters

### ISSUE THREAD

**Issue by Jean-Baptiste-Camps** [hipster-philology/pandora#23]
For now, when a model is saved, config.txt is just a copy of the original config.txt, instead of being saved from the parameters being used. This prevents, for instance, modifying a parameter during run (for instance a parameter counting the number of epochs ran so far, as needed for  #7  ). Here is the code:

        # save config file:
        if self.config_path:
            # make sure that we can reproduce parametrization when reloading:
            if not self.config_path == os.sep.join((self.model_dir, 'config.txt')):
                shutil.copy(self.config_path, os.sep.join((self.model_dir, 'config.txt')))
        else:
            with open(os.sep.join((self.model_dir, 'config.txt')), 'w') as F:
                F.write('# Parameter file\n\n[global]\n')
                F.write('nb_encoding_layers = '+str(self.nb_encoding_layers)+'\n')
                F.write('nb_dense_dims = '+str(self.nb_dense_dims)+'\n')
                F.write('batch_size = '+str(self.batch_size)+'\n')
                F.write('nb_left_tokens = '+str(self.nb_left_tokens)+'\n')
                F.write('nb_right_tokens = '+str(self.nb_right_tokens)+'\n')
                F.write('nb_embedding_dims = '+str(self.nb_embedding_dims)+'\n')
                F.write('model_dir = '+str(self.model_dir)+'\n')
                F.write('postcorrect = '+str(self.postcorrect)+'\n')
                F.write('nb_filters = '+str(self.nb_filters)+'\n')
                F.write('filter_length = '+str(self.filter_length)+'\n')
                F.write('focus_repr = '+str(self.focus_repr)+'\n')
                F.write('dropout_level = '+str(self.dropout_level)+'\n')
                F.write('include_token = '+str(self.include_context)+'\n')
                F.write('include_context = '+str(self.include_context)+'\n')
                F.write('include_lemma = '+str(self.include_lemma)+'\n')
                F.write('include_pos = '+str(self.include_pos)+'\n')
                F.write('include_morph = '+str(self.include_morph)+'\n')
                F.write('include_dev = '+str(self.include_dev)+'\n')
                F.write('include_test = '+str(self.include_test)+'\n')
                F.write('nb_epochs = '+str(self.nb_epochs)+'\n')
                F.write('halve_lr_at = '+str(self.halve_lr_at)+'\n')
                F.write('max_token_len = '+str(self.max_token_len)+'\n')
                F.write('min_token_freq_emb = '+str(self.min_token_freq_emb)+'\n')
                F.write('min_lem_cnt = '+str(self.min_lem_cnt)+'\n')
                F.write('curr_nb_epochs = '+str(self.curr_nb_epochs)+'\n') 

I suggest removing the 6 first lines.

**Comment by PonteIneptique** [comment]
Instead, I'd do

```python
    if not self.config_path or not self.config_path == os.sep.join((self.model_dir, 'config.txt'):
        # make sure that we can reproduce parametrization when reloading: 
        with open(os.sep.join((self.model_dir, 'config.txt')), 'w') as F:
            F.write('# Parameter file\n\n[global]\n')
            # ...
```
To ensure you do not overwrite your parameters. 

Maybe though, you should add a warning when the config of the model is part of the model dir. You do not want to overwrite you parameters with potentially new ones without noticing ;)

**Comment by Jean-Baptiste-Camps** [comment]
Actually, the idea is to include saving current configuration with saving the model. So, the idea is indeed to overwrite `./model/myModel/config.txt`, but not `./config.txt`.

**Comment by PonteIneptique** [comment]
If you were to remove the config file, you need to add an if, somewhere, which would raise an Error when `self.config_path == os.sep.join((self.model_dir, 'config.txt')`

**Comment by Jean-Baptiste-Camps** [comment]
I see your point, but, in practice, the new config.txt would differ only from the old (overwritten) one by the value of curr_nb_epochs. And, when you load a model, you have to specify the model/myModel/config.txt for loading to work. So, if I do like you suggest, when you load and retrain a model, you wouldn't be able to save config.txt.

If it really is necessary, I could do something like,

```python
if self.config_path == os.sep.join((self.model_dir, 'config.txt')
     shutil.copy(self.config_path, os.sep.join((self.model_dir, 'config_original.txt')))
```

**Comment by PonteIneptique** [comment]
That would be fine with me. But I'd add a warning message :)

**Comment by Jean-Baptiste-Camps** [comment]
Ok, then:

```python
if self.config_path == os.sep.join((self.model_dir, 'config.txt')
    shutil.copy(self.config_path, os.sep.join((self.model_dir, 'config_original.txt')))
    print('Warning: current config file will be overwritten. Saving it to config_original.txt')
```

**Comment by PonteIneptique** [comment]
Good to me ;)
But I'd probably warn before the whole trainer run

**Comment by PonteIneptique** [comment]
Fixed by #27
